chapter07/chapter07.adoc

```diff
@@ -0,0 +1,276 @@
+----
+import org.eclipse.microprofile.metrics.Metrics;
+import org.eclipse.microprofile.metrics.MetricType;
+
+public class ProductResource {
+
+    // Expose the invocation count as a counter metric
+    @Counted(name = "productAccessCount",
+        absolute = true,
+        description = "Number of times the list of products is requested")
+    public Response getProducts() {
+       // Method implementation
+       // ....
+    }
+}
+----
+  
+In the example above, the `@Counted` annotation tells MicroProfile Metrics to track the number of invocations of the `getProducts()` method and expose this metric as a counter. The name, and description of the metric can also be specified. 
+You can visit the following URL `++https://localhost:<port>/metrics?scope=application++` (Replace `<port>` with the actual port where the server is running) to see the number of times this method is called as below:
+
+[source]
+----
+…
+# HELP productAccessCount_total Number of times the list of products is requested
+# TYPE productAccessCount_total counter
+productAccessCount_total{mp_scope="application",} 3.0
+…
+----
+
+=== Creating a Custom Metric
+
+Creating a custom metric to track the number of products in a catalog involves using the MicroProfile Metrics API. This custom metric can be implemented as a gauge, which measures an instantaneous value (in this case, the current number of products in the catalog). 
+
+[source, java]
+----
+import org.eclipse.microprofile.metrics.annotation.Gauge;
+… 
+
+@Path("/products")
+@ApplicationScoped
+public class ProductResource {
+  // … 
+   
+  @GET
+  @Path("/count")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Gauge(name = "productCatalogSize", 
+    unit = "none", 
+  description = "Current number of products in the catalog")
+  public long getProductCount() {
+     return productCatalogSize;
+  }
+}
+----
+
+The gauge metric `productCatalogSize` can be accessed through the following endpoint: 
+
+`/metrics?name=io_microprofile_tutorial_store_product_resource_ProductResource_productCatalogSize`
+
+This custom metric implementation provides a real-time insight into the size of your product catalog, which can be invaluable for monitoring the scale of your service's data and understanding its behavior over time.
+
+=== Summary
+
+This Chapter delved into the intricacies of MicroProfile Metrics, illuminating its role as a pivotal specification for efficiently monitoring microservices. Now you are equipped with a thorough understanding of diverse metric types and their application for monitoring microservice performance. This chapter highlighted the need of regular microservice monitoring via metrics and health checks, emphasizing metrics for detailed performance insights such as response times and throughput. Through practical examples, this chapter showcases how to instrument microservices with MicroProfile Metrics, leveraging standard metrics, and creating custom metrics to monitor microservices comprehensively.
```

Comment:
need of regular -> need for regular

Reply:
Fixed with 4239e8c72e7b156e8433bfd440f8c5823707f76a